src/react/components/TriggerEditor/index.tsx

```diff
@@ -0,0 +1,544 @@
+    async (trigger: AstarteTrigger) => {
+      await handleFetchInterfacesName();
+      const interfaceName = _.get(trigger, 'simpleTriggers[0].interfaceName');
+      if (interfaceName === '*') {
+        return trigger;
+      }
+      const ifaceMajors = await handleFetchInterfaceMajors(interfaceName);
+      let ifaceMajor: number | undefined = _.get(trigger, 'simpleTriggers[0].interfaceMajor');
+      if (ifaceMajor == null) {
+        if (ifaceMajors.length === 0) {
+          return trigger;
+        }
+        ifaceMajor = Math.max(...ifaceMajors);
+        _.set(trigger as AstarteTrigger, 'simpleTriggers[0].interfaceMajor', ifaceMajor);
+      }
+      const interfaceMajor = ifaceMajor;
+      await handleFetchInterface({ interfaceName, interfaceMajor });
+      return trigger;
+    },
+    [handleFetchInterfacesName, handleFetchInterfaceMajors, handleFetchInterface],
+  );
+
+  const handleSimpleTriggerChange = useCallback((simpleTrigger: AstarteSimpleTrigger) => {
+    setTriggerDraft((draft) => ({ ...draft, simpleTriggers: [simpleTrigger] }));
+    if (simpleTrigger.type !== 'data_trigger' || simpleTrigger.interfaceName === '*') {
+      setTriggerInterface(null);
+    }
+  }, []);
+
+  const handleActionChange = useCallback(
+    (action: AstarteTrigger['action']) =>
+      setTriggerDraft((draft) => ({
+        ...draft,
+        action,
+      })),
+    [],
+  );
+
+  const handlePatchActionAmqpHeaders = useCallback(
+    (newHeaders: { [header: string]: string | undefined }) => {
+      setTriggerDraft((draft) => {
+        const oldAmqpStaticHeaders = _.get(draft.action, 'amqpStaticHeaders') || {};
+        const amqpStaticHeaders = _.omitBy(
+          {
+            ...oldAmqpStaticHeaders,
+            ...newHeaders,
+          },
+          _.isUndefined,
+        );
+        return {
+          ...draft,
+          action: {
+            ...draft.action,
+            amqpStaticHeaders,
+          },
+        };
+      });
+    },
+    [],
+  );
+
+  const handlePatchActionHttpHeaders = useCallback(
+    (newHeaders: { [header: string]: string | undefined }) => {
+      setTriggerDraft((draft) => {
+        const oldHttpStaticHeaders = _.get(draft.action, 'httpStaticHeaders') || {};
+        const httpStaticHeaders = _.omitBy(
+          {
+            ...oldHttpStaticHeaders,
+            ...newHeaders,
+          },
+          _.isUndefined,
+        );
+        return {
+          ...draft,
+          action: {
+            ...draft.action,
+            httpStaticHeaders,
+          },
+        };
+      });
+    },
+    [],
+  );
+
+  const handleTriggerNameChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setTriggerDraft((draft) => ({ ...draft, name: value }));
+  }, []);
+
+  const handleAddActionAmqpHeader = useCallback(() => {
+    setActiveModal({ modal: 'new-amqp-header' });
+  }, []);
+
+  const handleDismissNewAmqpHeaderModal = useCallback(() => {
```

Comment:
handleDismissNewAmqpHeaderModal, handleDismissEditAmqpHeaderModal and handleDismissNewHttpHeaderModal all act the same.
If we just need to dismiss the modal, we could define a `dismissModal` and reuse it.

Reply:
Fixed it.